velox/functions/lib/Re2Functions.cpp

```diff
@@ -454,20 +482,41 @@ class OptimizedLike final : public VectorFunction {
       EvalCtx& context,
       VectorPtr& resultRef) const final {
     VELOX_CHECK(args.size() == 2 || args.size() == 3);
+
+    bool isAscii = isAsciiArgs(rows, args);
```

Comment:
We don't need to check all arguments. Only the first one. And we don't need this check if Pattern is not kAtLeastN or kExactlyN.

Reply:
@mbasmanova What do you mean by "ascii flag is not yet available"?

Reply:
> auto isAscii = stringArg->isAscii(rows);

A vector stores is-ascii flag as metadata. The code above returns that stored value. However, if metadata hasn't been computed, then it will be missing. See SimpleVector.h (computeAndSetIsAscii).

Reply:
> Also, if it happens so that ascii flag is not yet available, it is cheaper to compute it on the fly then go through unicode slow path.

Hmm... I was trying to understand this sentence for quite a long time, 'then' should be 'than', right?

Reply:
Updated to only check ASCIIness for the first arg, and will compute ASCIIness on the fly if it is not available.